platform/iphone/gl_view.mm

```diff
@@ -379,369 +222,106 @@ - (void)destroyFramebuffer {
-	}
-	init_touches();
-	self.multipleTouchEnabled = YES;
-	self.autocorrectionType = UITextAutocorrectionTypeNo;
-
-	printf("******** adding observer for sound routing changes\n");
-	[[NSNotificationCenter defaultCenter]
-			addObserver:self
-			   selector:@selector(audioRouteChangeListenerCallback:)
-				   name:AVAudioSessionRouteChangeNotification
-				 object:nil];
-
-	printf("******** adding observer for keyboard show/hide\n");
-	[[NSNotificationCenter defaultCenter]
-			addObserver:self
-			   selector:@selector(keyboardOnScreen:)
-				   name:UIKeyboardDidShowNotification
-				 object:nil];
-	[[NSNotificationCenter defaultCenter]
-			addObserver:self
-			   selector:@selector(keyboardHidden:)
-				   name:UIKeyboardDidHideNotification
-				 object:nil];
-
-	//self.autoresizesSubviews = YES;
-	//[self setAutoresizingMask:UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleWidth];
-
-	return self;
-}
-
-//- (BOOL)automaticallyForwardAppearanceAndRotationMethodsToChildViewControllers {
-//	return YES;
-//}
-
-//- (BOOL)shouldAutorotateToInterfaceOrientation:(UIInterfaceOrientation)interfaceOrientation{
-//	return YES;
-//}
-
-// Stop animating and release resources when they are no longer needed.
-- (void)dealloc {
-	[self stopAnimation];
-
-	if ([EAGLContext currentContext] == context) {
-		[EAGLContext setCurrentContext:nil];
-	}
-
-	[context release];
-	context = nil;
-
-	[super dealloc];
-}
-
-- (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(NSDictionary *)change context:(void *)context {
-
-	if (object == _instance.avPlayerItem && [keyPath isEqualToString:@"status"]) {
-		if (_instance.avPlayerItem.status == AVPlayerStatusFailed || _instance.avPlayer.status == AVPlayerStatusFailed) {
-			_stop_video();
-			video_found_error = true;
-		}
-
-		if (_instance.avPlayer.status == AVPlayerStatusReadyToPlay &&
-				_instance.avPlayerItem.status == AVPlayerItemStatusReadyToPlay &&
-				CMTIME_COMPARE_INLINE(video_current_time, ==, kCMTimeZero)) {
-
-			//NSLog(@"time: %@", video_current_time);
-
-			[_instance.avPlayer seekToTime:video_current_time];
-			video_current_time = kCMTimeZero;
-		}
-	}
-
-	if (object == _instance.avPlayer && [keyPath isEqualToString:@"rate"]) {
-		NSLog(@"Player playback rate changed: %.5f", _instance.avPlayer.rate);
-		if (_is_video_playing() && _instance.avPlayer.rate == 0.0 && !_instance.avPlayer.error) {
-			dispatch_after(dispatch_time(DISPATCH_TIME_NOW, 0.5f * NSEC_PER_SEC), dispatch_get_main_queue(), ^{
-				[_instance.avPlayer play]; // NOTE: change this line according your current player implementation
-				NSLog(@"resumed play");
-			});
-
-			NSLog(@" . . . PAUSED (or just started)");
-		}
-	}
-}
-
-- (void)playerItemDidReachEnd:(NSNotification *)notification {
-	_stop_video();
-}
-
-/*
-- (void)moviePlayBackDidFinish:(NSNotification*)notification {
-
-
-		NSNumber* reason = [[notification userInfo] objectForKey:MPMoviePlayerPlaybackDidFinishReasonUserInfoKey];
```

Comment:
There are alternative APIs if we'd like to restore this functionality using non-deprecated classes.